h5i_image.go

```diff
@@ -0,0 +1,79 @@
+// license that can be found in the LICENSE file.
+
+package hdf5
+
+// #include "hdf5.h"
+// #include "hdf5_hl.h"
+// #include <stdlib.h>
+// #include <string.h>
+import "C"
+
+import (
+	"errors"
+	"image"
+	"image/color"
+	"unsafe"
+)
+
+// newImage takes a image object and convert it to a hdf5 format and write to the id node
+func newImage(id C.hid_t, name string, img image.Image) error {
+	if img == nil {
+		return errors.New("nil image!")
+	}
+	width := img.Bounds().Max.X
+	height := img.Bounds().Max.Y
+	var c_width, c_height C.hsize_t
+	c_name := C.CString(name)
+	defer C.free(unsafe.Pointer(c_name))
+	var gbuf []uint8
+	for y := 0; y < height; y++ {
+		for x := 0; x < width; x++ {
+			r, g, b, _ := img.At(x, y).RGBA()
+			gbuf = append(gbuf, uint8(r>>8))
```

Comment:
`gbuf = append(gbuf, uint8(r>>8), uint8(g>>8), uint8(b>>8))`, though I prefer `buf` and `byte`.